### PR TITLE
SFD-4097 Mock external API calls

### PR DESCRIPTION
**Pull request by daonhs**

### CODE REVIEW

**Review by ArifNhs — DISMISSED** [review]

File: build/automation/var/profile/dev.mk [hunk 1]
```diff
@@ -88,16 +88,11 @@ COGNITO_USER_POOL_CLIENT_ID := $(or $(COGNITO_USER_POOL_CLIENT_ID), )
 COGNITO_USER_POOL_ID := $(or $(COGNITO_USER_POOL_ID), )
 ADD_DEFAULT_COGNITO_USERS := false
 
-POSTCODE_MAPPING_SERVICE_URL := https://uec-dos-api-pca-dev-uec-dos-api-pc-ingress.k8s-nonprod.texasplatform.uk/api
+POSTCODE_MAPPING_SERVICE_URL := http://mockservice.sfs.test:8080/api
```
Comment (jope1) [inline build/automation/var/profile/dev.mk:91]: Is it possible to rename this URL so it's clearer what it is. Something like: mock.postcodeapi.sfs:8080/api ?

File: build/automation/var/profile/local.mk [hunk 2]
```diff
@@ -36,7 +36,7 @@ COGNITO_USER_POOL_CLIENT_ID := testUserPoolClientId
 COGNITO_USER_POOL_CLIENT_SECRET := testUserPoolClientSecret
 COGNITO_USER_POOL_ID := testUserPoolId
 
-POSTCODE_MAPPING_SERVICE_URL := https://uec-dos-api-pc-dev-uec-dos-api-pc-ingress.k8s-nonprod.texasplatform.uk/api
+POSTCODE_MAPPING_SERVICE_URL := http://mockservice.sfs.test:8080/api
```
Comment (jope1) [inline build/automation/var/profile/local.mk:39]: Is it possible to rename this URL so it's clearer what it is. Something like: mock.postcodeapi.sfs:8080/api ?

File: build/docker/docker-compose.yml [hunk 3]
```diff
@@ -48,13 +53,24 @@ services:
...
+    container_name: mockservice
+    volumes:
+      - $PROJECT_DIR/test/wiremock_mappings:/home/wiremock/mappings/
+    ports:
+      - "8090:8080"
+    command: [ "--verbose"]
+    networks:
+      default:
+        aliases:
+          - mockservice.sfs.test
```
Comment (jope1) [inline build/docker/docker-compose.yml:73]: Is it possible to rename this URL so it's clearer what it is. Something like: mock.postcodeapi.sfs:8080/api ?

File: build/docker/docker-compose.yml [hunk 4]
```diff
@@ -48,13 +53,24 @@ services:
...
+      - "9200:9200"
+    command: [ "elasticsearch" ]
     networks:
       default:
         aliases:
           - elasticsearch.sfs.local
-
+  mockservice:
+    image: rodolpheche/wiremock
+    container_name: mockservice
```
Comment (jope1) [inline build/docker/docker-compose.yml:64]: container_name: mock_postcodeapi ?

File: build/docker/docker-compose.yml [hunk 5]
```diff
@@ -48,13 +53,24 @@ services:
...
-      - "19200:9200"
-    command: ["elasticsearch"]
+      - "9200:9200"
+    command: [ "elasticsearch" ]
     networks:
       default:
         aliases:
           - elasticsearch.sfs.local
-
+  mockservice:
```
Comment (jope1) [inline build/docker/docker-compose.yml:62]: mock_postcodeapi ?

**Review by jope1 — CHANGES_REQUESTED** [review]

File: build/automation/var/profile/dev.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (daonhs) [reply build/automation/var/profile/dev.mk]: Created new jira ticket for this requirement

**Review by gwaardenburg — APPROVED** [review]

**Pull request merged**